### ISSUE TITLE
Uncaught SyntaxError: Invalid regular expression: /function\s **\s*\(|**\s*(=|:)\s*function\s*\(/...

### ISSUE THREAD

**Issue by codedaemon12** [DFreds/code-peek-atom#7]
[Enter steps to reproduce below:]
1. ...
2. ...

**Atom Version**: 1.8.0
**System**: Mac OS X 10.11.5
**Thrown From**: [code-peek](https://github.com/DFreds/code-peek-atom) package, v1.4.8
### Stack Trace

Uncaught SyntaxError: Invalid regular expression: /function\s **\s*(|**\s_(=|:)\s_function\s*(/: Nothing to repeat

```
At /Users/18740973/.atom/packages/code-peek/lib/supported-files.coffee:160

SyntaxError: Invalid regular expression: /function\s **\s*\(|**\s*(=|:)\s*function\s*\(/: Nothing to repeat
    at new RegExp (native)
    at Function.module.exports.SupportedFiles.getFunctionRegExpForFileType (/Users/18740973/.atom/packages/code-peek/lib/supported-files.coffee:160:16)
    at Object.module.exports.CodePeek.peekFunction (/Users/18740973/.atom/packages/code-peek/lib/code-peek.coffee:108:29)
    at atom-text-editor.subscriptions.add.atom.commands.add.code-peek:peekFunction (/Users/18740973/.atom/packages/code-peek/lib/code-peek.coffee:57:37)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:160:19)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchContextMenuCommand (/Applications/Atom.app/Contents/Resources/app.asar/src/atom-environment.js:1106:28)
    at EventEmitter.outerCallback (/Applications/Atom.app/Contents/Resources/app.asar/src/application-delegate.js:340:25)
    at emitThree (events.js:97:13)
    at EventEmitter.emit (events.js:175:7)
```
### Commands

```
     -1:11.1.0 core:save (atom-text-editor.editor.is-focused)
  3x -0:27.7.0 code-peek:peekFunction (atom-text-editor.editor.is-focused)
```
### Config

``` json
{
  "core": {
    "disabledPackages": [
      "linter-bootlint",
      "fonts",
      "symbols-view"
    ],
    "themes": [
      "one-dark-ui",
      "solarized-dark-syntax"
    ]
  }
}
```
### Installed Packages

``` coffee
# User
aligner, v0.22.0
aligner-css, v1.2.0
aligner-scss, v1.1.0
angularjs, v0.3.4
atom-beautify, v0.29.10
atom-ctags, v4.5.2
atom-html-preview, v0.1.21
atom-svg-icon-snippets, v0.3.3
auto-detect-indentation, v1.1.0
autoclose-html, v0.23.0
autocomplete-awk, v0.1.2
autocomplete-html-entities, v0.1.0
autocomplete-js-import, v1.0.1
autocomplete-json, v5.1.0
code-peek, v1.4.8
emmet, v2.4.3
file-icons, v1.7.17
file-watcher, v0.4.0
font-viewer, v0.3.1
fonts-cjk, v0.1.2
highlight-selected, v0.11.2
jQuery, v1.2.4
language-svg, v0.9.0
linter, v1.11.14
linter-jshint, v2.1.0
linter-jsonlint, v1.3.0
linter-sass-lint, v1.6.1
minimap, v4.24.7
pigments, v0.31.2
project-manager, v2.9.7
semanticolor, v0.2.2
svg-preview, v0.7.4
terminal-plus, v0.14.5

# Dev
No dev packages
```

**Comment by DFreds** [comment]
Hey, thanks for reporting this. Can you provide the language and the function definition you were trying to peek? I'll fix it if I get this information. I can't reproduce it otherwise.

Thanks!

**Comment by FlorianWendelborn** [comment]
@DFreds I think the cause of this issue is that the `REPLACE` string in [supported-files.coffee](https://github.com/DFreds/code-peek-atom/blob/8874ff32809fdf1c021e1b23a1171a8839235797/lib/supported-files.coffee#L153) isn't escaped. If you try to peek at a string that includes regular-expression special characters like `*`, this error can occur.

I managed to reproduce it in a `.js` file when peeking at `**`.

**Comment by DFreds** [comment]
Hmm, interesting. Okay, thank you for helping reproduce it. I'll look into it.

**Comment by DFreds** [comment]
Finally got around to this. Thanks for your input @dodekeract, you were right. Should be resolved in 1.4.11.
